dev-util/dwarves/files/dwarves-1.17-musl.patch

```diff
@@ -0,0 +1,190 @@
++)
+diff --git a/cmake/modules/FindOBSTACK.cmake b/cmake/modules/FindOBSTACK.cmake
+new file mode 100644
+index 0000000..905fc4c
+--- /dev/null
++++ b/cmake/modules/FindOBSTACK.cmake
+@@ -0,0 +1,19 @@
++# - Find obstack
++# If obstack is not implemented by libc find
++# a standalone version.
++
++message(STATUS "Checking availability of standalone obstack library")
++
++INCLUDE(CheckLibraryExists)
++
++find_path(OBSTACK_INCLUDE_DIR obstack.h
++	/usr/include
++	/usr/local/include
++	/usr/include/libdwarf
++	~/usr/local/include
++)
++
++find_library(OBSTACK_LIBRARY
++	NAMES obstack
++	PATHS /usr/lib /usr/local/lib /usr/lib64 /usr/local/lib64 ~/usr/local/lib ~/usr/local/lib64
++)
+diff --git a/dtagnames.c b/dtagnames.c
+index 0ffcbf7..1629fb3 100644
+--- a/dtagnames.c
++++ b/dtagnames.c
+@@ -14,9 +14,11 @@
+ 
+ static void print_malloc_stats(void)
+ {
++#ifdef __GLIBC__
+ 	struct mallinfo m = mallinfo();
+ 
+ 	fprintf(stderr, "size: %u\n", m.uordblks);
++#endif
+ }
+ 
+ static int class__tag_name(struct tag *tag, struct cu *cu __unused,
+diff --git a/dutil.h b/dutil.h
+index 0f5f74a..3d771c4 100644
+--- a/dutil.h
++++ b/dutil.h
+@@ -15,6 +15,7 @@
+ #include <elf.h>
+ #include <gelf.h>
+ #include <asm/bitsperlong.h>
++#include <linux/stddef.h>
+ #include "rbtree.h"
+ 
+ #define BITS_PER_LONG __BITS_PER_LONG
+diff --git a/dwarves.c b/dwarves.c
+index eb7885f..a437761 100644
+--- a/dwarves.c
++++ b/dwarves.c
+@@ -14,6 +14,7 @@
+ #include <fcntl.h>
+ #include <fnmatch.h>
+ #include <libelf.h>
++#include <limits.h>
+ #include <search.h>
+ #include <stdio.h>
+ #include <stdarg.h>
+diff --git a/dwarves_fprintf.c b/dwarves_fprintf.c
+index b8124a1..310631e 100644
+--- a/dwarves_fprintf.c
++++ b/dwarves_fprintf.c
+@@ -1910,7 +1910,11 @@ void cus__print_error_msg(const char *progname, const struct cus *cus,
+ void dwarves__fprintf_init(uint16_t user_cacheline_size)
+ {
+ 	if (user_cacheline_size == 0) {
++#ifdef __GLIBC__
+ 		long sys_cacheline_size = sysconf(_SC_LEVEL1_DCACHE_LINESIZE);
++#else
++		long sys_cacheline_size = 0;
++#endif
+ 
+ 		if (sys_cacheline_size > 0)
+ 			cacheline_size = sys_cacheline_size;
+diff --git a/dwarves_reorganize.c b/dwarves_reorganize.c
+index bae5b6e..62616f3 100644
+--- a/dwarves_reorganize.c
++++ b/dwarves_reorganize.c
+@@ -10,6 +10,7 @@
+ #include "list.h"
+ #include "dwarves_reorganize.h"
+ #include "dwarves.h"
++#include "dutil.h"
+ 
+ static void class__recalc_holes(struct class *class)
+ {
+diff --git a/hash.h b/hash.h
+index d3aa416..b1bd419 100644
+--- a/hash.h
++++ b/hash.h
+@@ -15,6 +15,7 @@
+  */
+ 
+ #include <stdint.h>
++#include <sys/reg.h>
+ 
+ /* 2^31 + 2^29 - 2^25 + 2^22 - 2^19 - 2^16 + 1 */
+ #define GOLDEN_RATIO_PRIME_32 0x9e370001UL
+diff --git a/lib/bpf/src/libbpf.c b/lib/bpf/src/libbpf.c
+index 514b1a5..b187734 100644
+--- a/lib/bpf/src/libbpf.c
++++ b/lib/bpf/src/libbpf.c
+@@ -32,6 +32,7 @@
+ #include <linux/filter.h>
+ #include <linux/list.h>
+ #include <linux/limits.h>
++#include <linux/stddef.h>
```

Comment:
Besides this, most of this seems upstreamable. The upstream linux patch didn't seem to go anywhere. See https://github.com/libbpf/libbpf/pull/41